webapp/src/components/call_widget/component.tsx

```diff
@@ -436,14 +436,14 @@ export default class CallWidget extends React.PureComponent<Props, State> {
             screenStream = window.callsClient.getLocalScreenStream();
         }
 
-        const wasRendering = Boolean(prevProps.screenSharingID && prevState.screenStream && prevProps.show);
-        const shouldRender = Boolean(this.props.screenSharingID && screenStream);
-
-        if (!this.state.screenStream && screenStream) {
+        const hasScreenTrackChanged = screenStream && this.state.screenStream && this.state.screenStream.getVideoTracks()[0].id !== screenStream.getVideoTracks()[0].id;
```

Comment:
Could this be simplified?
```suggestion
        const hasScreenTrackChanged = screenStream && this.state.screenStream?.getVideoTracks()[0].id !== screenStream.getVideoTracks()[0].id;
```

Reply:
I think so although it'd be purely a syntactic simplification rather than a logical one.